### ISSUE TITLE
Use global variables to share links and version across documentation

### ISSUE THREAD

**Issue by jooola** [LibreTime/libretime#1598]
**Is your feature request related to a problem? Please describe.**

There is no clean and easy way to share data using variables across the documentation files.

With mkdocs we had an `config` variable available at all time.
With jekyll we had a `site` variables available at all time.

With docusaurus we might be able to use mdx but it won't work inside code block or links, so links with a `{{ version }}` variable is not possible. But this is jsx, we have to import the the vars object in each file.

Maybe this require to write our own plugin.

@zklosko Could you please take a look ? I don't want to take care of this.

Docusaurus feels not really mature yet.

**Comment by zklosko** [comment]
The `docusaurus.config.js` file contains a collection of common variables (title, tagline, baseUrl). We can always have a `variables.js` file that we can import using JSX or MDX like you said. I don't see the need to have links with a version variable in them since the documentation is browsed by version set like a global. This could lead to 404s but I think there's a way to fallback to the latest version of the docs.

Was there a use other than versioning you were thinking of for global variables?

**Comment by jooola** [comment]
Maintaining a versioned documentation is too much work. 

I don't want to go though the entire pages to update a version number so people install from the latest version. This should be handled by variables. The jsx variables does not seem to be working in codeblocks.

This could be applied to other situation not just the libretime version number.


In addition I don't think the prose linter will understand the difference between jsx and normal text. This is why I would prefer not to use it too much.

**Comment by jooola** [comment]
Here are some examples:
```md
Download the [latest released](${vars.repository}/releases) tarball from Github.
```

```bash
wget https://github.com/libretime/libretime/releases/download/${vars.version}/libretime-${vars.version}.tar.gz
wget ${vars.tarball}
```

```bash
tar -xvf libretime-${vars.version}.tar.gz
cd libretime
```

**Comment by zklosko** [comment]
> The jsx variables does not seem to be working in codeblocks.

Codeblocks like this (ignore the `/`s) are literal:

```
/```
/```
```

Pulling in Docusaurus' `<CodeBlock>` component should allow us to use variables. Thanks for the examples.

**Comment by paddatrapper** [comment]
I don't think we should change documentation platforms yet again. If the variables don't work correctly, that seems to be an upstream issue that should be addressed by them. Any other missing features should be upstream feature requests

**Comment by zklosko** [comment]
@jooola See update above and this commit: https://github.com/zklosko/libretime/commit/f1156fd636241b6e47a4189d5cd5d23064992cdb

**Comment by jooola** [comment]
Fixed in #1611
